wgsl/index.bs

```diff
@@ -1183,15 +1564,13 @@ of a variable in [=address spaces/workgroup=] space.
 <div class='syntax' noexport='true'>
   <dfn for=syntax>array_type_decl</dfn> :
 
-    | [=syntax/array=] [=syntax/less_than=] [=syntax/type_decl=] ( [=syntax/comma=] [=syntax/element_count_expression=] ) ? [=syntax/greater_than=]
+    | [=syntax/array=] [=syntax/less_than=] [=syntax/type_decl=] ( [=syntax/comma=] [=syntax/shift_expression=] ) ? [=syntax/greater_than=]
```

Comment:
`shift_expression` here isn't great, but `expression` leads to a conflict with `relational_expression` which is probably unnecessary in this context.

Reply:
Looking at the 'expression' are two kinds of alternative grammar rules that can produce an integer:
* shift_expression
* bitwise expressions:
      * binary_and_expression and unary_expression
      * binary_or_expression or unary_expression
      * binary_xor_expression xor unary_expression


Instead of replacing element_count_expression with shift_expression change element_count_expression's definition:



     element_count_expression:
              shift_expression
            | bitwise_expression


    bitwise_expression : 
              binary_and_expression and unary_expression
            | binary_or_expression or unary_expression
            | binary_xor_expression xor unary_expression


and replace the last three options in `expression` to be bitwise_expression